src/haxelib/Data.hx

```diff
@@ -153,6 +153,8 @@ typedef Infos = {
 	@:optional var tags : Array<String>;
 	@:optional var dependencies : Dependencies;
 	@:optional var main:String;
+	@:optional var customDefines:String;
+	@:optional var customMetas:String;
```

Comment:
(sorry it seems I accidentally deleted my previous comment...) We can make this more concise by removing the "custom", as it's in the haxelib.json so it's clear that it's defined specifically for the library. We could also consider putting it in a common field, so it's cleaner if we add anything else:
```json
{
    "documentation" : {
        "metadata": "/path/to/json",
        "defines": "/path/to/json"
    }
}
```